src/lib/services/feature-toggle-service.ts

```diff
@@ -1357,6 +1363,52 @@ class FeatureToggleService {
         return fixedVariants;
     }
 
+    async crProtectedSaveVariantsOnEnv(
+        projectId: string,
+        featureName: string,
+        environment: string,
+        newVariants: IVariant[],
+        user: User,
+        oldVariants?: IVariant[],
+    ): Promise<IVariant[]> {
+        if (this.flagResolver.isEnabled('crOnVariants')) {
+            await this.stopWhenChangeRequestsEnabled(
+                projectId,
+                environment,
+                user,
+            );
+        }
+        return this.saveVariantsOnEnv(
+            projectId,
+            featureName,
+            environment,
+            newVariants,
+            user.username,
+            oldVariants,
+        );
+    }
+
+    async crProtectedSetVariantsOnEnvs(
+        projectId: string,
+        featureName: string,
+        environments: string[],
+        newVariants: IVariant[],
+        user: User,
+    ): Promise<IVariant[]> {
+        if (this.flagResolver.isEnabled('crOnVariants')) {
+            for (const env of environments) {
+                await this.stopWhenChangeRequestsEnabled(projectId, env);
+            }
+        }
```

Comment:
I'm thinking if we're missing something here... imagining we have CR enabled in some environment and we're trying to push certain variants configuration to all environments. This would require you to know which envs have change request and which don't... 

Perhaps this is something for the UI to split the Push into 2 sets, one for the CR endpoint another for the variants endpoint... Not convinced about this idea either

Reply:
Also, I think it would be nice to have a method that accepts all envs here, so we don't need to do n queries.